versions.lock

```diff
@@ -62,8 +62,8 @@ com.netflix.feign:feign-core:8.18.0 (3 constraints: de3f76e0)
 com.netflix.feign:feign-jackson:8.18.0 (1 constraints: c718909e)
 com.ning:compress-lzf:0.8.4 (1 constraints: 9a0f7888)
 com.oracle.database.jdbc:ojdbc11:21.5.0.0 (1 constraints: 98054146)
-com.palantir.cassandra:cassandra-all:2.2.13-1.6.0 (1 constraints: 5a067b5e)
-com.palantir.cassandra:cassandra-thrift:2.2.13-1.6.0 (2 constraints: 3f176da3)
+com.palantir.cassandra:cassandra-all:2.2.18-1.101.0 (1 constraints: bb06da6b)
+com.palantir.cassandra:cassandra-thrift:2.2.18-1.101.0 (2 constraints: 01186bf4)
```

Comment:
I'm not opposed to bumping dependencies here (https://github.com/palantir/atlasdb/blob/284da7ed1a404099ed4e6c6c7b0729c2d28fb5cf/atlasdb-cassandra/build.gradle#L128), its just that this hasn't been done in a very long time :)

Reply:
Also, does `recommendedProductDependencies` give us a strong guarantee that _if_ cassandra is deployed, it must be of this minimum version?

Reply:
Tbh this greatly simplifies the PR, and my original intention was that we wouldn't be blocked on rolling this out until C* has rolled out fleet-wide, but at this point it most likely will be, so this shouldn't cause/break any services from upgrading.

Reply:
That's what it's supposed to do - it's generally considered good enough that like if you have 404s or things like that because you tried to call some new API, we would view it as a thing that needed to be fixed in the infrastructure as opposed to the product (and yeah, the rule you describe _is_ what is desired in an optional dependency).

I think we are fine with a cassandra 3.31.0 (you have the code that handles the unknown methods) so no need to bump